### PR TITLE
Pr 405

### LINKED ISSUE #354: Deal with emojis on archived cc channels
Either strip on archive or allow on creation

### LINKED ISSUE #395: /send_notice should preserve formatting
Testing needed to reproduce bug.

Messages sent using this command should preserve all formating including channel names and line breaks etc. 

Probably should transition to a right-click message interaction as soon as we move to Discord.py 2.0.

### LINKED ISSUE #397: Add commands to toggle CC channels between public and private
Common request from CC owners is to make their channels private during setup and then make them public later. Probably should default new channels to private with a command usable in-channel by CC owners to make it public.

### LINKED ISSUE #399: Does case affect m.cc finding existing channels?
Why did rawmart_cc not get reused from this command? Was it the capital R confusing MAB?

`m.cc @Zergie Rawmart`

### LINKED ISSUE #405: Make send_notice usable by Cteam in CC channels

### LINKED ISSUE #425: New CC channels should be private
See also #397

### PR DESCRIPTION
**Pull request by cimspin**
This is an almost complete rework of the commands relating to CC channel management and the `/send_notice` function.

• `m.cc` - removed
• `m.cc_del` - removed
• "Community Carrier channel" renamed to "Community Channel", Community Carrier owner role will be renamed to Community Channel Owner.
• New command: `/create_community_channel`. Takes a user, a text string, and an optional emoji as arguments. Creates a Community Channel for the specified user, as `m.cc` used to. Fixes #354 
• New command: `/restore_community_channel` to re-use archived channels. Used in the target channel, takes a user as an argument. Fixes #399 
• Channels activated using the above commands will be private until opened (see below). Fixes #425 
• New commands: `/open_community_channel` and `/close_community_channel` commands to toggle a CC between public and private. Usable by Community Mentors and the Channel owner. Fixes #397 
• New command: `/remove_community_channel`. Used in the target channel to archive or delete it. If used outside a community channel, will run a check on the Community Carriers database and remove any orphaned channel owners. (If the delete option is used on a Community Channel, the channel is deleted after a 10-second timer. There's presently no lock function on this.)
• `/send_notice` is completely rewritten. It now uses a modal and returns an embed with optional title, 4000 character limit, and user display name/avatar. It will also optionally display an image via URL, and a thumbnail if defined (see below).
• `/send_notice` is now usable by Community Mentors as well as the channel owner - fixes #405 
• Added context menu app command: "Send CC Notice". Usable by Community Mentors and channel owners on any message in the target channel to rebroadcast it via MAB in a normal message (non-embed) with a channel role ping, similar to the old `/send_notice`. Will return an error if the target message has any role mentions or is too long for the bot to regurgitate.
• both forms of send_notice now preserve formatting - fixes #395 
• Added context menu app command: "Upload CC Thumb". Usable by Community Mentors and channel owners on any message in the target channel. If the message has an image attachment, it will upload as that channel's `send_notice` thumbnail. If not, any existing thumbnail will be deleted.
• New command: `/community_channel_help`. Usable by any Community team role, returns an ephemeral embed with a guide to all these commands.

Also, owing to some permission tweaks on the communtity channel categories, all community channels are now by default visible at all times to members of the Community team, whether archived or created-but-private.

### CODE REVIEW

**Comment by cimspin** [comment]
@skiedude @gcruicks I don't expect a review from either of you but will leave this up for a bit in case either of you want to look at it.

File: MissionAlertBot.py [hunk 1]
```diff
@@ -3528,22 +3463,153 @@ def check(message):
...
+    await interaction.followup.send(embed=embed)
+
+    return
+
+
+@bot.tree.command(name="create_community_channel",
+                  description="Create a Community Channel linked to a specific user.",
+                  guild=guild_obj)
+@check_roles([cmentor_role_id, botadmin_role_id])
+async def _create_community_channel(interaction: discord.Interaction, owner: discord.Member, channel_name: str, channel_emoji: str = None):
```
[the remark this answers is not in the document]
Comment (cimspin) [reply MissionAlertBot.py:3486]: If I'm understanding correctly, adding in the missing return on line 3500 should fix all these?

[the remark this answers is not in the document]
Comment (skiedude) [reply MissionAlertBot.py:3486]: yeah it should

**Review by skiedude — COMMENTED** [review]
Looks good

**Pull request merged**